runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/net/UrlParser.kt

```diff
@@ -23,16 +24,20 @@ internal fun urlParseImpl(url: String): Url =
             port = p ?: scheme.defaultPort
         }
 
+        val pathDecode: (String) -> String = when (decodingBehavior) {
+            UrlDecodingBehavior.DO_NOT_DECODE_PATH -> { s -> s }
+            UrlDecodingBehavior.DECODE_COMPONENTS -> String::urlDecodeComponent
+        }
         if (next.startsWith("/")) {
             next = next.capture(1 until next.firstIndexOrEnd("?", "#")) {
-                path = "/${it.urlDecodeComponent()}"
+                path = "/${pathDecode(it)}"
             }
         }
 
         if (next.startsWith("?")) {
             next = next.capture(1 until next.firstIndexOrEnd("#")) {
                 it.splitAsQueryString().entries.forEach { (k, v) ->
-                    parameters.appendAll(k.urlDecodeComponent(), v.map(String::urlDecodeComponent))
+                    parameters.appendAll(k.urlEncodeComponent(), v.map(String::urlDecodeComponent))
```

Comment:
**correctness**: Why are we now encoding the key?

Reply:
😳 That's an accidental change. Will fix.